### PR TITLE
[NeuralChat] support full parameters finetuning

### PR DESCRIPTION
**Pull request by lkk12014402**
## Type of Change

API changed for supporting full finetuning

## Description

add if branch for supporting full parameters finetuning

### CODE REVIEW

**Comment by lkk12014402** [comment]
add ut **WIP**

**Review by hshen14 — APPROVED** [review]

File: intel_extension_for_transformers/llm/finetuning/finetuning.py [hunk 1]
```diff
@@ -284,6 +284,15 @@ def finetune(self):
                 raise NotImplementedError(
                     f"Unsupported bits {finetune_args.bits}, only support 4 and 8 now."
                 )
+            if finetune_args.full_finetune:
+                raise ValueError(
+                    f"qlora and full_finetune can't be True at the same time."
+                )
+        elif finetune_args.full_finetune:
+            if finetune_args.bits in [4, 8]:
```
Comment (XinyuYe-Intel) [inline intel_extension_for_transformers/llm/finetuning/finetuning.py:292]:
```suggestion
            if finetune_args.bits not in [16, 32]:
```

Comment (lkk12014402) [reply intel_extension_for_transformers/llm/finetuning/finetuning.py]: fixed

**Comment by lkk12014402** [comment]
add ut done.

have tested lora/full_finetuning on gaudi/cpu/gpu
have tested qlora on gpu

**Review by XinyuYe-Intel — APPROVED** [review]

**Pull request merged**